tests: Fix issues in CentOS 8

xhernandez opened the pull request:
Due to some configuration changes in CentOS 8/RHEL 8, ssl-ciphers.t
and bug-1053579.t were failing.

The first one was failing because TLS v1.0 is disabled by default. The
test hash been updated to check that at least one of TLS v1.0, v1.1 or
v1.2 succeeds.

For the second case, the issue is that the test assumed that the
latest added group to a user should always be listed the last, but
this is not always true because nsswitch.conf now uses 'sss' before
'files', which means that data comes from a db that could not be
sorted.

Updates: #1009
Change-Id: I4ca01a099854ec25926c3d76b3a98072175bab06
Signed-off-by: Xavi Hernandez <xhernandez@redhat.com>

xhernandez committed:
tests: Fix issues in CentOS 8

Due to some configuration changes in CentOS 8/RHEL 8, ssl-ciphers.t
and bug-1053579.t were failing.

The first one was failing because TLS v1.0 is disabled by default. The
test hash been updated to check that at least one of TLS v1.0, v1.1 or
v1.2 succeeds.

For the second case, the issue is that the test assumed that the
latest added group to a user should always be listed the last, but
this is not always true because nsswitch.conf now uses 'sss' before
'files', which means that data comes from a db that could not be
sorted.

Updates: #1009
Change-Id: I4ca01a099854ec25926c3d76b3a98072175bab06
Signed-off-by: Xavi Hernandez <xhernandez@redhat.com>

xhernandez commented:
/run regression

mohit84 commented:
/run regression

mohit84 commented:
/recheck smoke

mykaul commented:
Below TLS 1.2 is considered insecure. We really need to disable it (and allow explicit override if needed?)
I've sent https://review.gluster.org/#/c/glusterfs/+/23957/ in the past but did not complete it.

xhernandez commented:
> Below TLS 1.2 is considered insecure. We really need to disable it (and allow explicit override if needed?)

I'm not against this, but given that the OS itself already has policies to decide what protocols are allowed, does it make sense to also implement that limitation in Gluster ?

The test failed because the OS has disabled those protocols, which seems good to me. If the administrator decides to enable them in the OS, currently Gluster will immediately be allowed to use them, without also having to configure Gluster. On the other side, if Gluster has an option to allow those protocols but the OS blocks them, they won't work, even if Gluster has been correctly configured.

I'm not sure if this adds any benefit to Gluster.

xhernandez committed:
tests: Fix TLS version detection

The old test didn't correctly determine which version of TLS should
be allowed by openssl.

Change-Id: Ic081c329d5ed1842fa9f5fd23742ae007738aec0
Signed-off-by: Xavi Hernandez <xhernandez@redhat.com>

xhernandez commented:
/run regression

amarts reviewed (approved):
I tend to agree with Xavi, glusterfs should use most of the system setting, and we should go with documenting how to change option if required. Many a times, system admins have their own reasons to have particular option, and we may have to honor it.

About this particular patch, as its helping 2 tests to pass regression. We should take it in.
